modules/auxiliary/scanner/mssql/mssql_version.rb

```diff
@@ -0,0 +1,58 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::MSSQL
+  include Msf::Auxiliary::Scanner
+  include Msf::OptionalSession::MSSQL
+
+  def initialize
+    super(
+      'Name' => 'MSSQL Version Utility',
+      'Description' => 'This module simply queries the MSSQL instance for version information.',
+      'Author' => 'MC',
+      'License' => MSF_LICENSE
+    )
+  end
+
+  def run
+    if session
+      set_session(session.client)
+    end
+    data = mssql_get_version
+    if data.nil? || data.empty?
+      print_error("Unable to retrieve version information for #{mssql_client.address}")
+      return
+    end
+
+    print_status("SQL Server for #{mssql_client.address}:")
+    if data['Version'] && !data['Version'].empty?
+      print_good("Version: #{data['Version']}")
+    else
+      print_error('Unknown Version')
+    end
+    if data['Encryption'] && !data['Encryption'].empty?
+      print_good("Encryption is #{data['Encryption']}")
+    else
+      print_error('Unknown encryption status')
+    end
+
+    report_mssql_service(mssql_client.address, data)
+  end
+
+  def report_mssql_service(ip, data)
+    mssql_info = 'Version: %<version>s, Encryption: %<encryption>s' % [
+      version: data['Version'] || 'unknown',
+      encryption: data['Encryption'] || 'unknown'
+    ]
+    report_service(
+      host: ip,
+      port: 1433,
```

Comment:
```suggestion
      port: rport,
```

Or if it's not accessible, `datastore['RPORT']`